### PR TITLE
Clean up and isolate js code in files select.html and listeners.html 

### LINKED ISSUE #1185: Transfer js code in select.html into separate .js file
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from underrepresented groups in free and open-source software!

We know that the process of creating a pull request is one of the biggest barriers for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Claim the issue by commenting. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

<details>
  <summary>💡 <i>Learn how to claim</i> 🙋</summary>
  <h3>Claiming an issue</h3>
  <p>Unless the issue is marked as reserved for someone, you can just say "I'd like to try this!" and then you've claimed it - no need to wait for someone to assign it to you. Just be sure you link your pull request (PR) to this issue so we can see where your solution is.</p>
  <p>And open one early if possible - even before you've completed it with additional commits - and others can help you figure out any issues you may face.</p>
</details>

- [ ] 📝 **Update** the file [select.html](https://github.com/publiclab/Leaflet.DistortableImage/blob/main/examples/select.html) in the `https://github.com/publiclab/Leaflet.DistortableImage` repository (press the little pen Icon) and edit the line as shown below.

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

In file select.html, insert the code `<script src="./js/select.js" defer></script>` in the header section and then apply the diff below.

```diff
-  <script>
-   let map;
-
-   (function(){
-     map = L.map('map').setView([51.505, -0.09], 13);
-     map.addGoogleMutant();
-
-     map.whenReady(function() {
-       img = L.distortableImageOverlay('example.jpg', {
-         corners: [
-           L.latLng(51.52, -0.14),
-           L.latLng(51.52,-0.10),
-           L.latLng(51.50, -0.14),
-           L.latLng(51.50,-0.10),
-         ],
-         mode: 'lock',
-       });
-
-       // create a second image
-       img2 = L.distortableImageOverlay('example.jpg', {
-         corners: [
-           L.latLng(51.51, -0.20),
-           L.latLng(51.51,-0.16),
-           L.latLng(51.49, -0.21),
-           L.latLng(51.49,-0.17),
-         ],
-         mode: 'freeRotate',
-         suppressToolbar: true,
-       });
-
-       img3 = L.distortableImageOverlay('example.jpg', {
-         corners: [
-           L.latLng(51.50, -0.13),
-           L.latLng(51.50,-0.09),
-           L.latLng(51.48, -0.14),
-           L.latLng(51.48,-0.10),
-         ],
-         actions: [L.DistortAction, L.FreeRotateAction, L.LockAction, L.OpacityAction, L.DeleteAction],
-       });
-
-       img4 = L.distortableImageOverlay('example.jpg', {
-         actions: [L.ScaleAction, L.DistortAction, L.RotateAction, L.FreeRotateAction, L.LockAction, L.OpacityAction, L.DeleteAction],
-         corners: [
-           L.latLng(51.51, -0.07),
-           L.latLng(51.51,-0.03),
-           L.latLng(51.49, -0.08),
-           L.latLng(51.49,-0.04),
-         ],
-       });
-
-       imgGroup = L.distortableCollection().addTo(map);
-
-       /* TODO: make an addLayers func */
-       imgGroup.addLayer(img);
-       imgGroup.addLayer(img2);
-       imgGroup.addLayer(img3);
-       imgGroup.addLayer(img4);
-     });
-   })();
- </script>
```
Create file `select.js` in folder `/examples/js`, then apply the diff below.
```diff
+    let map;
+
+    (function(){
+      map = L.map('map').setView([51.505, -0.09], 13);
+      map.addGoogleMutant();
+
+      map.whenReady(function() {
+        img = L.distortableImageOverlay('example.jpg', {
+          corners: [
+            L.latLng(51.52, -0.14),
+            L.latLng(51.52,-0.10),
+            L.latLng(51.50, -0.14),
+            L.latLng(51.50,-0.10),
+          ],
+          mode: 'lock',
+        });
+
+        // create a second image
+        img2 = L.distortableImageOverlay('example.jpg', {
+          corners: [
+            L.latLng(51.51, -0.20),
+            L.latLng(51.51,-0.16),
+            L.latLng(51.49, -0.21),
+            L.latLng(51.49,-0.17),
+          ],
+          mode: 'freeRotate',
+          suppressToolbar: true,
+        });
+
+        img3 = L.distortableImageOverlay('example.jpg', {
+          corners: [
+            L.latLng(51.50, -0.13),
+            L.latLng(51.50,-0.09),
+            L.latLng(51.48, -0.14),
+            L.latLng(51.48,-0.10),
+          ],
+          actions: [L.DistortAction, L.FreeRotateAction, L.LockAction, L.OpacityAction, L.DeleteAction],
+        });
+
+        img4 = L.distortableImageOverlay('example.jpg', {
+          actions: [L.ScaleAction, L.DistortAction, L.RotateAction, L.FreeRotateAction, L.LockAction, L.OpacityAction, L.DeleteAction],
+          corners: [
+            L.latLng(51.51, -0.07),
+            L.latLng(51.51,-0.03),
+            L.latLng(51.49, -0.08),
+            L.latLng(51.49,-0.04),
+          ],
+        });
+
+        imgGroup = L.distortableCollection().addTo(map);
+
+        /* TODO: make an addLayers func */
+        imgGroup.addLayer(img);
+        imgGroup.addLayer(img2);
+        imgGroup.addLayer(img3);
+        imgGroup.addLayer(img4);
+      });
+    })();
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

(This issue was created by [First-Timers-Bot](https://github.com/hoodiehq/first-timers-bot).)

### PR DESCRIPTION
**Pull request by segun-codes**
This PR replaces PRs #1280 by @ayushkumarbhadani, #1274 & #1278 by @adityagoyal200, #1254 and #1255 by @Leeiaisabelle & #1275 by @adityagoyal200.

Fixes #1185 and #1184 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updates
* [ ] @mention the original creator of the issue in a comment below for help or for a review

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by segun-codes** [comment]
Hi @jywarren, kindly review and consider for merging, many thanks!

This PR replaces PRs #1280 by @ayushkumarbhadani, #1274 & #1278 by @adityagoyal200, #1254 and #1255 by @Leeiaisabelle which were at various levels of completion and required further actions from affected contributors but the common denominator is that none was given final landing.

The task of closing out with this PR is as agreed with @7malikk.

**Comment by 7malikk** [comment]
Great work @segun-codes

**Comment by segun-codes** [comment]
> Great work @segun-codes

Thanks!!!

File: examples/js/select.js [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+    imgGroup = L.distortableCollection().addTo(map);
+
+    /* TODO: make an addLayers func */
+    imgGroup.addLayer(img);
+    imgGroup.addLayer(img4);
+    imgGroup.addLayer(img2);
+    imgGroup.addLayer(img3);
+    });
+})();
+=======
```
Comment (jywarren) [inline examples/js/select.js:59]: Oops looks like a leftover from git here!

File: src/edit/iconsets/KeymapperIconSet.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+L.KeymapperIconSet = L.IconSet.extend({
+
+  _symbols:
+      // eslint-disable-next-line max-len
```
Comment (jywarren) [inline src/edit/iconsets/KeymapperIconSet.js:4]: Not a big deal but since you'll make another commit, would you mind using two space indents everywhere you can? It helps to slowly make the codebase consistent. CC @7malikk as well, it's not critical but nice to have a standard. Thanks!

**Review by jywarren — APPROVED** [review]
Looks great, thanks! Left a couple small requests.

**Comment by jywarren** [comment]
And many thanks to @ayushkumarbhadani, @adityagoyal200, and @Leeiaisabelle for their work on these, which is being combined here! ❤️ 🎉

File: examples/js/select.js (hunk 1 above)
Comment (segun-codes) [reply examples/js/select.js]: @jywarren this is now addressed, many thanks!

File: src/edit/iconsets/KeymapperIconSet.js (hunk 2 above)
Comment (segun-codes) [reply src/edit/iconsets/KeymapperIconSet.js:4]: @jywarren, absolutely I will be happy to stick to this rather nice standard. I believe @7malikk will do same. Many thanks!

**Comment by segun-codes** [comment]
@jywarren, do you want to review and merge now so we call it day on this?

**Review by jywarren — APPROVED** [review]
Looks super. Thanks!!
